internal/app/servicequotas/servicequotas.go

```diff
@@ -58,29 +64,44 @@ func (s *serviceQuotaFetcher) GetStatistics() Statistics {
 
 // GetQuota retrieves and returns the AWS quota value for the specified serviceCode and quotaCode
 func (s *serviceQuotaFetcher) getQuota(serviceCode string, quotaCode string) (float64, error) {
+	_, span := tracer.Start(s.ctx, "get-quota")
+	defer span.End()
+
+	span.SetAttributes(attribute.String("com.qonto.prometheus_rds_exporter.aws.quota.service_code", serviceCode), attribute.String("com.qonto.rds_exporter.aws.quota.code", quotaCode))
+
 	params := &aws_servicequotas.GetServiceQuotaInput{
 		ServiceCode: &serviceCode,
 		QuotaCode:   &quotaCode,
 	}
 
 	s.statistics.UsageAPICall++
 
-	result, err := s.client.GetServiceQuota(context.TODO(), params)
+	result, err := s.client.GetServiceQuota(s.ctx, params)
 	if err != nil {
+		span.SetStatus(codes.Error, "failed to get quota")
+		span.RecordError(err)
+
 		return 0, fmt.Errorf("can't get %s/%s service quota: %w", serviceCode, quotaCode, err)
 	}
 
 	// AWS response payload could contains errors (eg. missing permission)
 	if result.Quota.ErrorReason != nil {
 		s.logger.Error("AWS quota error: ", "errorCode", result.Quota.ErrorReason.ErrorCode, "message", *result.Quota.ErrorReason.ErrorMessage)
+		span.SetStatus(codes.Error, "failed to fetch quota")
+		span.RecordError(err)
```

Comment:
As I understand, `err` is not being filled here (we'll probably have `null`).

Maybe we can create the err variable with what we're logging a few lines above

Reply:
Good catch! I updated it to use the already exisint `errQuotaError`